src/server/pachyderm_test.go

```diff
@@ -297,6 +298,36 @@ func TestPipeline(t *testing.T) {
 	buffer = bytes.Buffer{}
 	require.NoError(t, pfsclient.GetFile(pachClient, outRepo.Name, outCommits[0].Commit.ID, "file", 0, 0, "", nil, &buffer))
 	require.Equal(t, "foo\nbar\n", buffer.String())
+
+	require.NoError(t, ppsclient.DeletePipeline(pachClient, pipelineName))
+
+	pipelineInfos, err := pachClient.ListPipeline(context.Background(), &ppsclient.ListPipelineRequest{})
+	require.NoError(t, err)
+	for _, pipelineInfo := range pipelineInfos.PipelineInfo {
+		require.True(t, pipelineInfo.Pipeline.Name != pipelineName)
+	}
+
+	// Do third commit to repo; this time pipeline should not run since it's been deleted
+	commit3, err := pfsclient.StartCommit(pachClient, dataRepo, commit2.ID, "")
+	require.NoError(t, err)
+	_, err = pfsclient.PutFile(pachClient, dataRepo, commit3.ID, "file", strings.NewReader("buzz\n"))
+	require.NoError(t, err)
+	require.NoError(t, pfsclient.FinishCommit(pachClient, dataRepo, commit3.ID))
+
+	// We will sleep a while to wait for the pipeline to actually get cancelled
+	// Also if the pipeline didn't get cancelled (due to a bug), we sleep a while
+	// to let the pipeline commit
+	time.Sleep(5 * time.Second)
```

Comment:
Bummer that we have to do a sleep here... but I guess there's no way to wait for something to not happen.